lib/liquid/document.rb

```diff
@@ -25,5 +36,19 @@ def unknown_tag(tag, parse_context)
         raise SyntaxError, parse_context.locale.t("errors.syntax.unknown_tag", tag: tag)
       end
     end
+
+    def render_to_output_buffer(context, output)
+      @body.render_to_output_buffer(context, output)
+    end
+
+    def render(context)
+      @body.render(context)
+    end
+
+    private
+
+    def new_body
+      Liquid::BlockBody.new
+    end
```

Comment:
This repeats the implementation in `Liquid::Block`. If the two are always intended to do the same thing / be overridden together, how about we expose a single implementation somewhere higher up? (for example, `Liquid.new_block_body`) This way Liquid C only needs to override one method.

Reply:
Pushed a commit that extracted the implementation to override into `ParseContext#new_block_body`.  I'll update the liquid-c PR to override that.